### ISSUE TITLE
Compiler warning on Windows with latest rtools42

### ISSUE THREAD

**Issue by tlapak** [Rdatatable/data.table#5413]
Documenting because I just saw this, but I haven't looked into it yet.

Building data.table with the latest version of rtools42 (5253-5107) on Windows and under the latest version of R 4.2.1 produces the following warning

```
In file included from fread.h:3,
                 from fread.c:27:
dt_stdio.h:25: warning: "__USE_MINGW_ANSI_STDIO" redefined
   25 |   #define __USE_MINGW_ANSI_STDIO 1
      | 
In file included from c:\rtools42\x86_64-w64-mingw32.static.posix\include\corecrt.h:10,
                 from c:\rtools42\x86_64-w64-mingw32.static.posix\include\crtdefs.h:10,
                 from c:\rtools42\x86_64-w64-mingw32.static.posix\include\time.h:9,
                 from c:\rtools42\x86_64-w64-mingw32.static.posix\include\sys\time.h:10,
                 from fread.c:8:
c:\rtools42\x86_64-w64-mingw32.static.posix\include\_mingw.h:435: note: this is the location of the previous definition
  435 | #define __USE_MINGW_ANSI_STDIO 0      /* was not defined so it should be 0 */
      | 
```

**Comment by tdhock** [comment]
duplicate of #5374, fixed by #5395

**Comment by tlapak** [comment]
Ah thanks. Sorry that I missed that!
